test/unit/org/apache/cassandra/Util.java

```diff
@@ -819,4 +820,38 @@ public static void setUpgradeFromVersion(String version)
                                                    VersionedValue.unsafeMakeVersionedValue(version, v + 1));
         Gossiper.instance.expireUpgradeFromVersion();
     }
+
+    public interface ThrowingRunnable
+    {
+        void run() throws Throwable;
+    }
+
+    @SuppressWarnings("ConstantConditions")
+    public static void retry(ThrowingRunnable runnable, Predicate<Throwable> canRetry, int maxRetries, long delay) throws Throwable
```

Comment:
very much like Awaitability.

can we get a more informative method name please.

Reply:
It is perfectly applicable...